### PR TITLE
Feature/6.3/tcomp 372

### PR DESCRIPTION
**Pull request by sgandon**
Initial PR from wwang
- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TCOMP-372

**What is the new behavior?**
https://jira.talendforge.org/browse/TCOMP-372

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

**Other information**:
https://jira.talendforge.org/browse/TCOMP-372

### CODE REVIEW

**Comment by sgandon** [comment]
I have merge master onto this branch and updated the getSample code from the DatasetRuntime to match the new interface method signature.
I have also added a helper class to adapt any Source Reader to a getSample new behavior, see ReaderDataProvider (for the lack of a better name).

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1136/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1136/)

File: components-api/src/test/java/org/talend/components/api/component/runtime/ReaderDataProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.joda.time.Instant;
+import org.junit.Test;
+import org.talend.daikon.exception.TalendRuntimeException;
+import org.talend.daikon.java8.Consumer;
+
+public class ReaderDataProviderTest {
```
Comment (gonchar-ivan) [inline components-api/src/test/java/org/talend/components/api/component/runtime/ReaderDataProviderTest.java]: I like it!

**Comment by gonchar-ivan** [comment]
LGTM

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1161/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1161/)

**Pull request merged**